src/tsserverWeb/tsconfig.json

```diff
@@ -0,0 +1,16 @@
+{
+    "extends": "../tsconfig-library-base",
+    "compilerOptions": {
+        "outDir": "../../built/local",
+        "rootDir": ".",
+        "target": "esnext",
+        "module": "esnext",
+        "lib": ["esnext"],
+        "declaration": false,
+        "sourceMap": true,
+        "tsBuildInfoFile": "../../built/local/tsserverWeb.tsbuildinfo"
+    },
+    "files": [
+        "tsserverWeb.ts",

```

Comment:
_tsserverWeb.js_ only includes a patch for untransformed dynamic `import()`. We can't embed that directly in _tsserver_ as it would result in `import()` being transformed into a `require()`. _webServer.ts_ conditionally loads _tsserverWeb.js_ based on whether it is being run from a Web Worker and whether that Web Worker supports dynamic `import()` syntax.

If we moved all of _webServer.ts_ into _tsserverWeb_, then you wouldn't be able to run _tsserver_ in the browser if that browser for some reason didn't support dynamic `import()`. This instead isolates that specific syntax to a single file that is conditionally imported via `importScripts()`.